### PR TITLE
[Tools] GstPipe<-->Pbtxt Conversion / Skeleton

### PR DESCRIPTION
**Pull request by myungjoo**
Skeleton code for GstPipe<-->Pbtxt Conversion.

Ultimately, we can use web-based WYSIWYG tools based on this!

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: tools/development/gstPrototxt.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        if name is None or len(name) == 0:
+            raise RuntimeError("Incorrect usage of __getFullPadname.")
+        if padname is None or len(padname) == 0:
+            return name + "."
+        else:
+            return name + "." + padname
+
+    ## @brief Verify filter and return object if filter is given with name
+    def __verifyfilter(f):
+        _a = a
```
Comment (dongju-chae) [inline tools/development/gstPrototxt.py:90]: Where does `a` come from?

Comment (myungjoo) [reply tools/development/gstPrototxt.py:90]: Yes.. it should've been f.

File: tools/development/gstPrototxt.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+##
```
Comment (kparichay) [inline tools/development/gstPrototxt.py:2]: for scripts, SPDX can be at max at line 2 (https://lore.kernel.org/patchwork/patch/883424/)

File: tools/development/gstPrototxt.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+# @todo   WIP
+#
+
+## @todo Get python pbtxt parser
+
+## @todo property link up tensor-repo-*
+
+declaredNames = [] # list of names declared by user.
+generatedNameCounter = 0
+generatedNamePrefix = '__id'
```
Comment (kparichay) [inline tools/development/gstPrototxt.py:23]: this is a global constant than global variable right? How about CAPITALIZING it?

File: tools/development/gstPrototxt.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+        self.element = elementname
+        if _name is not None and len(_name) > 0:
+            name = _name
+            if name in declaredNames:
+                raise RuntimeError("Duplicated name in elements: " + str(name))
+            declaredNames.append(str(name))
+        else:
+            while ((generatedNamePrefix + str(generatedNameCounter)) not in declaredNames):
+                generatedNameCounter = generatedNameCounter + 1
```
Comment (kparichay) [inline tools/development/gstPrototxt.py:47]: this increases the name counter for differently named variables. Is this intended?

File: tools/development/gstPrototxt.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            name = generatedNamePrefix + str(generatedNameCounter)
+            nameGenerated = True
```
Comment (kparichay) [inline tools/development/gstPrototxt.py:49]: bring this outside the `if else` and add this name to `declaredNames`?

File: tools/development/gstPrototxt.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        self.sinks = []
+
+    ## @brief add a filter instnace
+    def addFilter(_filter):
+        self.filters.append(_filter)
+        self.nameList.append(_filter.name)
+
+    ## @brief Get filter with the given name
+    def __getFilterByName(name):
+        for f in self.filters:
```
Comment (kparichay) [inline tools/development/gstPrototxt.py:74]: in python , its recommended to use `filter` than use a loop.
also, as nameList stores all the names of filters, checking directly in nameList would also suffice.

File: tools/development/gstPrototxt.py (hunk 3 above)
Comment (myungjoo) [reply tools/development/gstPrototxt.py:23]: Applied.

File: tools/development/gstPrototxt.py (hunk 2 above)
Comment (myungjoo) [reply tools/development/gstPrototxt.py:2]: Applied.

File: tools/development/gstPrototxt.py (hunk 5 above)
Comment (myungjoo) [reply tools/development/gstPrototxt.py:49]: No, the intention is to manage filters with its id declared (```filtername id=THENAME```) and filters without such (generated) differently. If it is generated (not declared), we can hide the id in GUI as well. If it is declared, we need to show the id in GUI.

File: tools/development/gstPrototxt.py (hunk 6 above)
Comment (myungjoo) [reply tools/development/gstPrototxt.py:74]: Updated.

**Review by dongju-chae — APPROVED** [review]
LGTM, after this PR is merged, I will take over this topic.

**Review by anyj0527 — APPROVED** [review]
LGTM 👍

**Pull request merged**